web/src/stream_settings_ui.js

```diff
@@ -721,36 +712,15 @@ export function change_state(section, right_side_tab) {
         return;
     }
 
-    if (section === "subscribed") {
-        toggler.goto("subscribed");
-        stream_edit.empty_right_panel();
-        return;
-    }
-
     // if the section is a valid number.
     if (/\d+/.test(section)) {
         const stream_id = Number.parseInt(section, 10);
-        const sub = sub_store.get(stream_id);
-        // There are a few situations where we can't display stream settings:
-        // 1. This is a stream that's been archived. (sub=undefined)
-        // 2. The stream ID is invalid. (sub=undefined)
-        // 3. The current user is a guest, and was unsubscribed from the stream
-        //    stream in the current session. (In future sessions, the stream will
-        //    not be in sub_store).
-        //
-        // In all these cases we redirect the user to 'subscribed' tab.
-        if (!sub || (page_params.is_guest && !stream_data.is_subscribed(stream_id))) {
-            toggler.goto("subscribed");
-            stream_edit.empty_right_panel();
-        } else {
-            show_right_section();
-            stream_edit_toggler.set_select_tab(right_side_tab);
-            switch_to_stream_row(stream_id);
-        }
```

Comment:
Have removed this checks from here as we do it in `hashchange.js` and `hash_util.js`, but we might need to take care of not calling this function without validating the url in future.

Reply:
Seems not hard to avoid, sounds good.